mp4parse_capi/src/lib.rs

```diff
@@ -741,13 +741,12 @@ fn create_sample_table(track: &Track) -> Option<Vec<mp4parser_indice>> {
                 }
             true
         });
-    if !ret {
+    if sample_table.is_empty() {
```

Comment:
Oops, you're right! I worry about `.collect()` being inefficient when we don't need the resulting vector. `.count()` might be more semantically-neutral than `.all()` or `.any()`.

I asked on irc; #rust-beginners recommends using a for loop instead of map().something() when the primary result is a side-effect.